data/src/main/java/com/microsoft/azure/kusto/data/ClientFactory.java

```diff
@@ -73,7 +75,9 @@ public static StreamingClient createStreamingClient(ConnectionStringBuilder csb)
      * @throws URISyntaxException if the cluster URL is invalid
      */
     public static StreamingClient createStreamingClient(ConnectionStringBuilder csb, HttpClientProperties properties) throws URISyntaxException {
-        return new ClientImpl(csb, properties);
+        HttpClientProperties httpClientProperties = Optional.ofNullable(properties)
```

Comment:
I still think that's a bit bad, since it's a performance hit (creating a new object), while not providing much readability over a simple ternary.

I know it's copied from another place, but change it there too?

Reply:
OK
This is a very scala like code- so i guess im used to it - in scala you are encouraged to use this and dont have ternary 
I dont see how creating a new object on a 1 time created client is a performance hit.